Zach log in page

ZachDFK opened the pull request:
Combining lasted branches together to see if we can make them fit together

ZachDFK committed:
made a working very basic sign up page

ZachDFK committed:
made a working very basic sign up page

ZachDFK committed:
formating issues fix

ZachDFK committed:
formating issues fix

ZachDFK committed:
used validation methods in user model now uses has_secure_password to protect password in db

ZachDFK committed:
used validation methods in user model now uses has_secure_password to protect password in db

GoodOlChuck committed:
adding basic post and comment functionality

GoodOlChuck committed:
adding basic post and comment functionality

GoodOlChuck committed:
add static home page and messing around with html styles

GoodOlChuck committed:
add static home page and messing around with html styles

ZachDFK committed:
login page and functionalbility

ZachDFK committed:
login page and functionalbility

ZachDFK committed:
fixing test cases

ZachDFK committed:
fixing test cases

GoodOlChuck commented on app/views/sessions/home.html.erb:
```diff
@@ -0,0 +1,5 @@
+<p id='notice'> <%=notice%></p>
+
+<%= link_to 'Login', login_path %>
+<%= link_to 'Singup', signup_path %>
```
Small typo 'Singup'

GoodOlChuck commented on app/views/users/_signup_form.html.erb:
```diff
@@ -0,0 +1,34 @@
+    </div>
+  <% end %>
+
+<% end %>
```
There is no newline at the end of this file.

ZachDFK committed:
Update login.html.erb

ZachDFK committed:
Update login.html.erb

ZachDFK committed:
Update home.html.erb

ZachDFK committed:
Update home.html.erb

ZachDFK committed:
Update home.html.erb

ZachDFK committed:
format and typos

ZachDFK committed:
Gemfil.lock commit for new gem(should of been done earlier)

ZachDFK committed:
commit with all working parts for a very simple demo

ZachDFK committed:
commit with all working parts for a very simple demo

ZachDFK committed:
attempt to past posts tests

ZachDFK committed:
attempt to past posts tests

ZachDFK committed:
temporary removal of test for post while we figure a way to test it

ZachDFK committed:
temporary removal of test for post while we figure a way to test it

kevincox commented on Gemfile:
```diff
@@ -12,6 +12,7 @@ gem 'sass-rails', '~> 5.0'
 gem 'therubyracer', platforms: :ruby
 gem 'turbolinks'
 gem 'uglifier', '>= 1.3.0'
+gem 'bcrypt'
```
I still think that it is weird that you have to do this but whatever.

kevincox commented on app/controllers/application_controller.rb:
```diff
@@ -3,7 +3,7 @@ class ApplicationController < ActionController::Base
- end
+ 	def hello
+		render text:"hi bud"
+	end
```
I think you used tabs here while the rest of the file is spaces.

kevincox commented on app/models/user.rb:
```diff
@@ -1,16 +1,45 @@
-  validates :email, uniqueness: true, presence: true
+has_many :posts
+has_many :comments
+require 'bcrypt'
```
Fix the indent please.

kevincox commented on app/views/posts/_form.html.erb:
```diff
@@ -21,7 +21,7 @@
   <div class="field">
     <%= f.label :author %><br>
-    <%= f.number_field :author %>
+    <%= f.label User.get_username_by_id(@post.author) %>
```
Is this temporary? Why do they need to specify the author in the form? Shouldn't it use the logged in user?

kevincox commented on app/views/posts/index.html.erb:
```diff
@@ -17,7 +17,7 @@
         <td><%= post.title %></td>
         <td><%= post.contents %></td>
-        <td><%= post.author %></td>
+        <td><%= User.get_username_by_id(post.author) %></td>
```
Shouldn't you be able to do `post.author.username` or something?

kevincox commented:
Some more tests would be nice. For example a login and logout test.

kevincox commented on app/models/user.rb:
```diff
@@ -1,16 +1,45 @@
+		else
+			return id
+		end
+	end
```
This method shouldn't be necessary. Something like `User.find_by_id(whatever).try :username` will work. I don't see the value in this abstraction.

kevincox commented on app/models/user.rb:
```diff
@@ -1,16 +1,45 @@
+		else
+			user = User.find_by_username(username_or_email)
+		end
+		if user && user.authenticate(login_password)
```
Style nitpick.

``` rb
if user && user.authenticate(login_password)
   user
end
```

If you want to be extra snazzy use try.

``` rb
if user.try :authenticate, login_password
   user
end
```

I can't wait for the elvis operator to ship.

``` rb
if user.?authenticate login_password
   user
end
```

kevincox commented on app/models/user.rb:
```diff
@@ -1,16 +1,45 @@
+			return id
+		end
+	end
+	def self.authenticate_with_username_or_email(username_or_email="",login_password="")
```
Why do these have default arguments? I can't see a single use case of not passing them in. If there is a use case we can add them later.

ZachDFK committed:
early morning commit

ZachDFK committed:
early morning commit

kevincox committed:
Always have pp (pretty-print) available.

kevincox committed:
Merge pull request #19 from kevincox/kevincox-pp

Always have pp (pretty-print) available.

ZachDFK committed:
late morning commit

ZachDFK committed:
late morning commit

ZachDFK committed:
late morning commit

ZachDFK committed:
late morning commit

kevincox commented on app/controllers/comments_controller.rb:
```diff
@@ -1,7 +1,9 @@
-
+  	
   	@comment = @post.comments.create(comment_params)
   	redirect_to post_path(@post)
```
You're mixing tabs and spaces in this file.

GoodOlChuck committed:
Update readme for milestone 2

GoodOlChuck committed:
Fix typo

kevincox committed:
Fail CI on tabs.

As much as I <3 tabs rails generates everything without them and we have
to pick one.

ZachDFK committed:
last commit before milesstone 2

ZachDFK committed:
last commit before milesstone 2

kevincox committed:
Merge pull request #20 from kevincox/kevincox-no-tabs

Fail CI on tabs.

GoodOlChuck committed:
update

ZachDFK committed:
fixed route merging issue

ZachDFK committed:
fixed route merging issue

kevincox committed:
Print line number of tabs.

ZachDFK committed:
space to tab fix

ZachDFK committed:
space to tab fix

kevincox committed:
Add editor config.

kevincox committed:
Merge pull request #21 from kevincox/kevincox-line-number

Better tab stuff.

ZachDFK committed:
fixed route error

ZachDFK committed:
fixed route error

ZachDFK committed:
commit to check with maste
